CONTENTdm.js

```diff
@@ -0,0 +1,577 @@
+		];
+  for ( var i in replace ) {
+	for ( var j in replace[ i ] ) {
+	  if ( item[ replace[ i ][ 0 ] ] ) break;
+	  else {
+		item[ replace[ i ][ 0 ] ] = item[ replace[ i ][ j ] ];
+		if ( j > 0 ) item[ replace[ i ][ j ] ] = "";
+	  }
+	}
+  }
+}
+
+function getTitle( doc, docTitle, url ) {
+  var title;
+  if ( docTitle.match( "::" ) ) docTitle = docTitle.substring( 0, docTitle.indexOf( "::" ) );
+  if ( url.match( "/cdm/landingpage" ) ) title = docTitle + ": About this collection";
+  else title = docTitle; // Fallback: first part of webpage title
+  return title.replace( /^\s+|\s+$/g, '' );
+}
+
+function cleanAuthor( item ) {
+  var authorType = [ "creator", "contributor" ]; // Author field names in the mapping table must be Zotero author types
+  for ( var i in authorType ) {
+	if ( item[ authorType[ i ] ] ) {
+	  fetchedAuthor = item[ authorType[ i ] ].replace( /\([^)]*\)|\[[^\]]*\]|\;$/g, '' ).replace( /, Sir |, Dr /g, ', ' ); // Fetch and remove parentheses and honorifics
+	  var nameArray = [];
+	  item[ authorType[ i ] ] = "";
+	  nameArray = fetchedAuthor.split( ";" ); // Split authors at semicolon and push into an array
+	  for ( var j in nameArray ) item.creators.push( ZU.cleanAuthor( nameArray[ j ], authorType[ i ], 1 ) );
+	}
+  }
+  for ( var k in item.creators ) {
+	if ( !item.creators[ k ].firstName ) item.creators[ k ].fieldMode = 1;
+  }
+}
+
+function attachPDF( doc, item ) {
+  var pdfPath = '//embed';
+  var pdfScrape = ZU.xpath( doc, pdfPath );
+  var pdfObject = pdfScrape[ 0 ]; // Could iterate looking for multiple attachments
+  var pdfTitle;
+  if ( pdfObject && pdfObject.src ) { // Can't simply test for .src
+	var itemTitle = fetchXPathText( doc, '//*[@id="metadata_title"]' );
+	if ( itemTitle ) pdfTitle = "Attachment: " + itemTitle;
+	else pdfTitle = "Attachment";
+	item.attachments.push( {
+	  url: pdfObject.src,
+	  title: pdfTitle,
+	  mimeType: "application/pdf"
+	} );
+  }
+}
+
+function doWeb( doc, url ) {
+  if ( detectWeb( doc, url ) == "multiple" ) { // If on a multiple result page, call Zotero.selectItems
+	var titlePath = '//div/ul/li/ul/li[2]/div/a';
+	var items = {};
+	var articles = [];
+	var titles = ZU.xpath( doc, titlePath );
+	if ( titles.length < 1 ) return false; // Null search results will be ignored
+	for ( var i in titles ) {
+	  items[ titles[ i ].href ] = ZU.trimInternal( titles[ i ].textContent );
+	}
+	Zotero.selectItems( items, function( items ) {
+	  if ( !items ) return true;
+	  for ( var i in items ) {
+		articles.push( i );
+	  }
+	  ZU.processDocuments( articles, scrape );
+	} );
+
+  } else if ( url.match( "/cdm/fullbrowser" ) ) {
+	var originatingItem = ZU.xpath( doc, '//link[@rel="canonical"]' ); // Look for link to originating item
+	if ( originatingItem[ 0 ] && originatingItem[ 0 ].href ) {
+	  url = originatingItem[ 0 ].href;
+	  ZU.processDocuments( url, scrape );
+	} else return false;
+
+  } else if ( ZU.xpath( doc, '//input[@id="cdm_newRefUrl"]' ) ) {
```

Comment:
( `ZU.xpath().length`  would be the way to go==> it's 0 for a null object, which evaluates to false)